libs/blocks/global-navigation/utilities/keyboard/index.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable class-methods-use-this */
+import { getNextVisibleItem, getPreviousVisibleItem, selectors } from './utils.js';
+import MainNav from './mainNav.js';
+import { closeAllDropdowns } from '../utilities.js';
+
+const cycleOnOpenSearch = ({ e, isDesktop }) => {
+  const withoutBreadcrumbs = [
+    ...document.querySelectorAll(`
+  ${selectors.brand}, 
+  ${selectors.mainNavToggle},
+  ${selectors.mainNavItems},
+  ${selectors.searchTrigger},
+  ${selectors.searchField},
+  ${selectors.signIn},
+  ${selectors.profileButton},
+  ${selectors.logo}
+  `),
+  ];
+  const first = getNextVisibleItem(-1, withoutBreadcrumbs);
+  const last = getPreviousVisibleItem(withoutBreadcrumbs.length, withoutBreadcrumbs);
+  const openSearch = isDesktop && document.querySelector(selectors.openSearch);
+  if (openSearch && document.activeElement === withoutBreadcrumbs[e.shiftKey ? first : last]) {
+    e.preventDefault();
+    withoutBreadcrumbs[e.shiftKey ? last : first].focus();
```

Comment:
You're using this 2 times, maybe it could be stored in a variable?

Reply:
Its two different conditions, that look similar `first : last` and `last: first`.. but with extra `if's` it becomes a lot worse, so that was the best way. (Or maybe I didn't get your point and need an example 😁)